src/table/dtable.jl

```diff
@@ -0,0 +1,110 @@
+import Tables
+import TableOperations
+
+import Base:fetch
+
+export DTable
+
+const VTYPE = Vector{Union{Dagger.Chunk,Dagger.EagerThunk}}
+
+"""
+    struct DTable
+
+Structure representing the distributed table based on Dagger.
+
+The table is stored as a vector of `Chunk` structures which hold partitions of the table.
+That vector can also store `EagerThunk` structures when an operation that modifies
+the underlying partitions was applied to it (currently only `filter`).
+
+Underlying partitions should always be `DataFrame` structures.
+"""
+struct DTable
+    chunks::VTYPE
+    tabletype
+    DTable(chunks::VTYPE, tabletype) = new(chunks, tabletype)
+    DTable(chunks::Vector{Dagger.EagerThunk}, args...; kwargs...) = new(VTYPE(chunks), args...; kwargs...)
+    DTable(chunks::Vector{Dagger.Chunk}, args...; kwargs...) = new(VTYPE(chunks), args...; kwargs...)
+end
+
+include("iterators.jl")
+include("operations.jl")
+
+""" 
+    DTable(table, chunksize) -> DTable
+
+Constructs a `DTable` using a `Tables.jl` compatible `table` input.
+It assumes no initial partitioning of the table and uses the `chunksize`
+argument to partition the table (based on row count).
+"""
+function DTable(table, chunksize::Integer;tabletype=nothing)
+    if !Tables.istable(table)
+        throw(ArgumentError("Provided input is not Tables.jl compatible."))
+    end
+    parts = Tables.partitions(TableOperations.makepartitions(table, chunksize))
+    chunks = Vector{Dagger.Chunk}()
+    sizehint!(chunks, length(parts))
+    _transform = Tables.materializer(table)
+    if tabletype !== nothing
+        _transform = Tables.materializer(tabletype())
+    end
+    append!(chunks, Dagger.tochunk.(_transform.(parts)))
+    return DTable(chunks, if tabletype === nothing typeof(table) else tabletype end)
```

Comment:
```suggestion
    return DTable(chunks, tabletype === nothing ? typeof(table) : tabletype)
```

It's mostly just preference, but I prefer a ternary expression when it's all on one line and very simple.